hermes/probe/create_file.go

```diff
@@ -0,0 +1,124 @@
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	// "os"
+	// "sync"
+
+	// "cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	// "github.com/google/cloudprober/logger"
+	// "github.com/google/cloudprober/metrics"
+	// "github.com/google/cloudprober/probes/options"
+	// "github.com/google/cloudprober/targets/endpoint"
+	// "github.com/googleinterns/step224-2020/blob/master/hermes/hermes.go"
+)
+
+const (
+	begin = 1  // ID of the first HermesFile
+	end   = 50 // ID of the last HermesFile
+	maxFileSize = 1000 // maximum allowed file size in bytes 
+)
+
+type RandomHermesFile struct {
+	ID   int64 // ID is a positive integer in the range [1,50]
+	Size int   // File size in bytes
+}
+
+type randomHermesFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+func (r *randomHermesFileReader) readDone() bool {
+	return r.i >= r.size
+}
+
+func (r *randomHermesFileReader) bytesLeft() int {
+	return (r.size - r.i)
+}
+
+func (r *randomHermesFileReader) bufferTooLong(buffer []byte) bool {
+	return len(buffer) > r.bytesLeft()
+}
+
+func (r *randomHermesFileReader) Read(buff []byte) (n int, err error) {
+	if r.readDone() {
+		return 0, io.EOF
+	}
+	b := buff
+	if r.bufferTooLong(b){
+		b = buff[:r.bytesLeft()]
+	}
+	n, err = r.rand.Read(b)
+	if(err!=nil){
+		return 0, err // 0 as nil casn't be returned as a type int argument
+	}
+	r.i += n
+	return
+}
+
+// Warning: NewReader is not thread safe
+func (f *RandomHermesFile) NewReader() *randomHermesFileReader {
+	return &randomHermesFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.ID))} // ID will serve as a Seed and i will be set to 0 automatically
+}
+
+func (f *RandomHermesFile) CheckSum() ([]byte, error) {
```

Comment:
Is it expected that the checksum will only be calculated once during the lifetime of the object?
If it's needed more than once, it would better to store the result instead of re-calculating it.

Reply:
The checksum will be calculated once when the object is created and stored in the string form in the fileName.
It will then in readFile be calculated again and compared with the one stored in the string form in the fileName to make sure that the contents of the file haven't changed

Reply:
Do you mean the checksum will be recovered from the filename by parsing the string?
I would advise against that.